qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteDiagnostics.java

```diff
@@ -1090,4 +1100,64 @@ private ResolvedJavaTypeInfo validateIterable(Part part, Section ownerSection,
 		}
 		return resolvedJavaType;
 	}
-}
+
+	private void validateSwitchWhenPart(JavaMemberInfo member, Section ownerSection, String projectUri,
+			ResolvingJavaTypeContext resolvingJavaTypeContext, PropertyPart propertyPart, MethodPart methodPart,
+			List<Diagnostic> diagnostics) {
+		String javaTypeParent = member.getJavaElementType();
+		String partName = "";
+		if (methodPart != null) {
+			partName = methodPart.toString();
+		} else if (propertyPart != null) {
+			partName = propertyPart.toString();
+		}
+		List<Node> children = ownerSection.getChildren();
+		// Iterate through children and ensure the type of the #case or #is parts are
+		// the same as the parent section
+		for (Node node : children) {
+			if (node.getKind() == NodeKind.Section) {
+				List<Parameter> value = new ArrayList<Parameter>();
+				if (ownerSection.getSectionKind() == SectionKind.SWITCH) {
+					value = ((CaseSection) node).getParameters();

```

Comment:
Here you can have ClassCastException if you your section is not a #case (ex : #is). The main idea is to check if the current child node is a #is or #case otherwise you do nothing. I suggest that you do:

I suggest that you do:

```java
for (Node childNode : children) {
	if (childNode.getKind() == NodeKind.Section) {
		Section childSection = (Section) childNode;
		if (!(childSection.getSectionKind() == SectionKind.CASE || childSection.getSectionKind() == SectionKind.IS)) {
			// the current child section is not a #id or #case, ignore it 
			continue;
		}
		List<Parameter> value = childSection.getParameters();
```